lib/validators/serialtiles.js

```diff
@@ -21,20 +22,73 @@ module.exports = function validateSerialtiles(opts, callback) {
     .on('tile', function(t) {
       if (errored) return;
       if (!t.buffer) return;
-      if (t.buffer.length <= limits.max_tilesize) return;
-      callback(invalid('Tile exceeds maximum size of ' + Math.round(limits.max_tilesize/1024) + 'k at z' + t.z + '. Reduce the detail of data at this zoom level or omit it by adjusting your minzoom.'));
-      errored = true;
-      read.close();
+
+      var gzipped = (tiletype.headers(t.buffer))['Content-Encoding'] === 'gzip';
+      var format = tiletype.type(t.buffer);
+
+      // Not gzipped and not a valid tiletype
+      if (!gzipped && (format !== 'png' && format !== 'webp' && format !== 'jpg')) {
+        callback(invalid('Invalid tiletype'));
+        errored = true;
+        read.close();
+      }
+      
+      // Gzipped pbf
+      if (gzipped) {
+        //gunzip buffer and validate
+        validateGzipped(t, function(err) {
+          if (err) {
+            callback(invalid('Invalid pbf: %s', err));
+            errored = true;
+            read.close();
+          }
+
+          if (t.buffer.length <= limits.max_tilesize) return;
+          else callback(invalid('Tile exceeds maximum size of ' + Math.round(limits.max_tilesize/1024) + 'k at z' + t.z + '. Reduce the detail of data at this zoom level or omit it by adjusting your minzoom.'));
+          errored = true;
```

Comment:
Was thinking of just splitting this out into a error/close function...but that doesn't cut the clutter down too much to be worth it. If I think of something better, will implement.